pkgs/stdenv/generic/stdenv.sh

```diff
@@ -0,0 +1,63 @@
+# This file is only used for shellcheck, not actually sourced via bash.
+# The name of the file makes it, so that it can be included via directive like this:
+#   # shellcheck source=stdenv.sh
+#
+# Shellcheck has rule SC2154 "var is referenced but not assigned":
+#   https://www.shellcheck.net/wiki/SC2154
+# Naturally, this rule is hard to validate for all our setup hooks, because they
+# depend on variables:
+#   - set in stdenv/generic/setup.sh
+#   - set in default setup-hooks defined in stdenv/generic/default.nix (.../build-support/setup-hooks/...)
+#   - passed as derivation arguments
+#
+# We can tell shellcheck about "dependencies" indirectly, with the following construct:
+#   # shellcheck source=setup.sh
+#   . /dev/null
+# This will source /dev/null, so nothing. The shellcheck meta command will replace this
+# with sourcing the generic/setup.sh file. Thus, all variables defined in stdenv will be
+# known to shellcheck.
+
+# All paths are relative to pkgs/stdenv/generic, thanks to .shellcheckrc.
+# shellcheck source=setup.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/audit-tmpdir.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/compress-man-pages.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/make-symlinks-relative.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/move-docs.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/move-lib64.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/move-sbin.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/move-systemd-user-units.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/multiple-outputs.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/patch-shebangs.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/prune-libtool-files.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/reproducible-builds.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/set-source-date-epoch-to-latest.sh
+. /dev/null
+# shellcheck source=../../build-support/setup-hooks/strip.sh
+. /dev/null
```

Comment:
ShellCheck's `source` directive searches the paths relative to the "source paths", which defaults to the **current working directory** of ShellCheck.

Fortunately, we could specify the source path using the `source-path` directive for each `source` directive and use the special value `SCRIPTDIR` to represent the script's `basedir`.

See https://www.shellcheck.net/wiki/Directive

```suggestion
# shellcheck source-path=SCRIPTDIR
# shellcheck source=setup.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=audit-tmpdir.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=compress-man-pages.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=make-symlinks-relative.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=move-docs.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=move-lib64.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=move-sbin.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=move-systemd-user-units.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=multiple-outputs.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=patch-shebangs.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=prune-libtool-files.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=reproducible-builds.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=set-source-date-epoch-to-latest.sh
. /dev/null
# shellcheck source-path=SCRIPTDIR/../../build-support/setup-hooks
# shellcheck source=strip.sh
. /dev/null
```

Reply:
Oops! I didn't notice that `.shellcheckrc`.